test/unit/wifi.js

```diff
@@ -109,182 +104,352 @@ exports['Tessel.prototype.findAvailableNetworks'] = {
-        test.deepEqual(self.simpleExec.args[2][0], commands.reconnectWifi());
-        test.equal(self.logsInfo.calledOnce, true);
-        test.equal(self.logsInfo.args[0][1].indexOf('Enabled.') !== -1, true);
+      .then(() => {
+        test.equal(this.simpleExec.calledThrice, true);
+        test.deepEqual(this.simpleExec.args[0][0], commands.turnOnWifi(state));
+        test.deepEqual(this.simpleExec.args[1][0], commands.commitWirelessCredentials());
+        test.deepEqual(this.simpleExec.args[2][0], commands.reconnectWifi());
+        test.equal(this.logsInfo.calledTwice, true);
+        test.equal(this.logsInfo.args[0][1].indexOf('Enabled.') !== -1, true);
+        test.ok(this.ubusListen.callCount, 1);
         test.done();
       })
       .catch(function(err) {
-        test.equal(undefined, err, 'an unexpected error was thrown');
+        test.fail(err);
+        test.done();
       });
+  },
+  setWifiStateFalsy: function(test) {
+    test.expect(7);
+    var state = false;
 
-    setImmediate(function() {
-      // enable wifi completed
-      self.tessel._rps.emit('close');
-      setImmediate(function() {
-        // commit wifi settings completed
-        self.tessel._rps.emit('close');
-        setImmediate(function() {
-          // Reconnecting to wifi completed
-          self.tessel._rps.emit('close');
+    // Test is expecting several closes...;
+    this.tessel._rps.on('control', (command) => {
+      if (command.toString() === 'ubus listen' && state) {
+        // Write to stdout so it completes as expected
+        // Wrap in setImmediate to make sure listener is set up before emitting
+        setImmediate(() => {
+          this.tessel._rps.stdout.emit('data', 'ifup');
         });
-      });
+      } else {
+        setImmediate(() => {
+          // Remove any listeners on stdout so we don't break anything when we write to it
+          this.tessel._rps.stdout.removeAllListeners();
+
+          // Continue
+          this.tessel._rps.emit('close');
+        });
+      }
     });
-  },
-  setWifiStateFalsy: function(test) {
-    var self = this;
 
-    test.expect(6);
-    var state = false;
     this.tessel.setWiFiState(state)
-      .then(function() {
-        test.equal(self.simpleExec.calledThrice, true);
-        test.deepEqual(self.simpleExec.args[0][0], commands.turnOnWifi(state));
-        test.deepEqual(self.simpleExec.args[1][0], commands.commitWirelessCredentials());
-        test.deepEqual(self.simpleExec.args[2][0], commands.reconnectWifi());
-        test.equal(self.logsInfo.calledOnce, true);
-        test.equal(self.logsInfo.args[0][1].indexOf('Disabled.') !== -1, true);
+      .then(() => {
+        test.equal(this.simpleExec.calledThrice, true);
+        test.deepEqual(this.simpleExec.args[0][0], commands.turnOnWifi(state));
+        test.deepEqual(this.simpleExec.args[1][0], commands.commitWirelessCredentials());
+        test.deepEqual(this.simpleExec.args[2][0], commands.reconnectWifi());
```

Comment:
The way they're set up now, it looks like we're testing `simpleExec` instead of the functionality of `setWifiState`. 

Would it be clearer to use the same pattern of spying on the commands and making sure they were called in the function rather than checking their output? That way we can still ensure that the right methods are called in the flow of `setWifiState`.